xieyu_wang/Graph.cpp

```diff
@@ -56,7 +56,11 @@ bool Graph::isConnected(int fromVertexId, int toVertexId) {
   assert(hasVertex(toVertexId) && "Vertex doesn't exist");
   for (const auto& fromVertexEdgeId : getVertex(fromVertexId).getEdgeIds()) {
     for (const auto& toVertexEdgeId : getVertex(toVertexId).getEdgeIds()) {
-      if (fromVertexEdgeId == toVertexEdgeId && fromVertexId != toVertexId) {
+      if (fromVertexEdgeId == toVertexEdgeId) {
+        const auto edge = getEdge(fromVertexEdgeId);
```

Comment:
Why do you make copy here
```suggestion
        const auto& edge = getEdge(fromVertexEdgeId);
```